app/main.py

```diff
@@ -1,13 +1,36 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"], "name": error["code"],
+            "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {"line": error["line_number"],
+             "column": error["column_number"],
+             "message": error["text"],
+             "name": error["code"], "source": "flake8"}
+            for error in errors
+        ],
+        "path": file_path,
+        "status": "failed",
+    }
```

Comment:
The logic for formatting each error in `format_single_linter_file` is identical to that in `format_linter_error`. Consider using `format_linter_error` within this function to avoid code duplication and improve maintainability.